### PR TITLE
Split download service into web and publishing

### PR DESCRIPTION
**Pull request by mattrout92**
### What

- Split service into web and publishing
- Add authenticate function to determine authentication status of caller/user
- Add middleware

### How to review

- Verify logic is correct

### Who can review

Team B

### CODE REVIEW

File: dp-download-service.nomad [hunk 1]
```diff
@@ -117,7 +117,7 @@ job "dp-download-service" {
       }
 
       vault {
-        policies = ["dp-download-service", "read-psk"]
```
Comment (CarlHembrough) [inline dp-download-service.nomad]: Is the `read-psk` no longer required here?

File: handlers/download.go [hunk 2]
```diff
@@ -51,15 +52,17 @@ type Download struct {
...
 
-func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
+func setStatusCode(req *http.Request, w http.ResponseWriter, err error, logData log.Data) {
 	status := http.StatusInternalServerError
 	message := internalServerMessage
 	if err, ok := err.(ClientError); ok {
 		status = err.Code()
 	}
-	log.ErrorR(req, err, log.Data{"setting-response-status": status})
+	logData["setting-response-status"] = status
```
Comment (CarlHembrough) [inline handlers/download.go]: Snake case for logs I think ?

File: dp-download-service.nomad (hunk 1 above)
Comment (mattrout92) [reply dp-download-service.nomad]: The content of "read-psk" has been moved to the dp-download-service-publishing policy

File: handlers/download.go [hunk 3]
```diff
@@ -147,3 +154,32 @@ func (d Download) Do(extension string) http.HandlerFunc {
...
+	if d.IsPublishing {
+		var hasCallerIdentity, hasUserIdentity bool
+
+		callerIdentity := identity.Caller(r.Context())
+		if callerIdentity != "" {
+			logData["caller_identity"] = callerIdentity
+			hasCallerIdentity = true
+		}
+
+		userIdentity := identity.User(r.Context())
```
Comment (CarlHembrough) [inline handlers/download.go]: I don't think you need to check for a user identity here. The caller identity will contain either the calling service or calling user. The user identity can be populated from an incoming request header so is not safe to use to determine authentication. See the `identity.isPresent(ctx)` function in go-ns, as it does what you need in this function : https://github.com/ONSdigital/go-ns/blob/b7c32727ee0de2d1e987539302b141890d287b6a/identity/authentication.go#L35

File: handlers/download.go [hunk 4]
```diff
@@ -147,3 +154,32 @@ func (d Download) Do(extension string) http.HandlerFunc {
...
 }
+
+func (d Download) authenticate(r *http.Request, logData map[string]interface{}) (bool, map[string]interface{}) {
+	var authorised bool
+
+	if d.IsPublishing {
+		var hasCallerIdentity, hasUserIdentity bool
+
+		callerIdentity := identity.Caller(r.Context())
+		if callerIdentity != "" {
```
Comment (CarlHembrough) [inline handlers/download.go]: If you decide not to use the `identity.isPresent(ctx)` function - consider also checking for a nil value here

**Pull request merged**